src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -522,6 +522,28 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 			break;
 		}
 
+		case T_CreateStatsStmt:
+		{
+			CreateStatsStmt *createStatsStmt = (CreateStatsStmt *) parseTree;
+			RangeVar *relation = linitial(createStatsStmt->relations);
+
+			char **relationName = &(relation->relname);
+			char **objectSchemaName = &(relation->schemaname);
+
+			SetSchemaNameIfNotExist(objectSchemaName, schemaName);
+			AppendShardIdToName(relationName, shardId);
+
+			Value *statNameValue = llast(createStatsStmt->defnames);
+			char *statName = strVal(statNameValue);
+			char **statNameVar = &(statName);
+			AppendShardIdToName(statNameVar, shardId);
+			statNameValue = makeString(pstrdup(statName));
+			createStatsStmt->defnames = list_make2(makeString(pstrdup(schemaName)),
+												   statNameValue);
```

Comment:
Using the schemaName would cause statistics to always be in the same schema with the table but they don't have to be.

You can also add some related tests.

Reply:
Good point. Fixed. Thanks